lib/omnibus/packagers/fastmsi.rb

```diff
@@ -0,0 +1,654 @@
+    #
+    # @param [String] glob
+    #   a ruby acceptable glob path such as with **, *, [] etc.
+    #
+    # @return [String] path relative to the project's install_dir
+    #
+    # Raises exception the glob matches 0 or more than 1 file/directory.
+    #
+    def gem_path(glob = NULL)
+      unless glob.is_a?(String) || null?(glob)
+        raise InvalidValue.new(:glob, 'be an String')
+      end
+
+      install_path = Pathname.new(project.install_dir)
+
+      # Find path in which the Chef gem is installed
+      search_pattern = install_path.join('**', 'gems')
+      search_pattern = search_pattern.join(glob) unless null?(glob)
+      file_paths  = Pathname.glob(search_pattern).find
+
+      raise "Could not find `#{search_pattern}'!" if file_paths.none?
+      raise "Multiple possible matches of `#{search_pattern}'! : #{file_paths}" if file_paths.count > 1
+      file_paths.first.relative_path_from(install_path).to_s
+    end
+    expose :gem_path
+
+    #
+    # @!endgroup
+    # --------------------------------------------------
+
+    # @see Base#package_name
+    def package_name
+      bundle_msi ? bundle_name : msi_name
+    end
+
+    def msi_name
+      "#{project.package_name}-#{project.build_version}-#{project.build_iteration}-#{Config.windows_arch}.msi"
+    end
+
+    def bundle_name
+      "#{project.package_name}-#{project.build_version}-#{project.build_iteration}-#{Config.windows_arch}.exe"
+    end
+
+    #
+    # The path where the MSI resources will live.
+    #
+    # @return [String]
+    #
+    def resources_dir
+      File.expand_path("#{staging_dir}/Resources")
+    end
+
+    #
+    # Write the localization file into the staging directory.
+    #
+    # @return [void]
+    #
+    def write_localization_file
+      render_template(resource_path('localization-en-us.wxl.erb'),
+        destination: "#{staging_dir}/localization-en-us.wxl",
+        variables: {
+          name:          project.package_name,
+          friendly_name: project.friendly_name,
+          maintainer:    project.maintainer,
+        }
+      )
+    end
+
+    #
+    # Write the parameters file into the staging directory.
+    #
+    # @return [void]
+    #
+    def write_parameters_file
+      render_template(resource_path('parameters.wxi.erb'),
+        destination: "#{staging_dir}/parameters.wxi",
+        variables: {
+          name:            project.package_name,
+          friendly_name:   project.friendly_name,
+          maintainer:      project.maintainer,
+          upgrade_code:    upgrade_code,
+          parameters:      parameters,
+          version:         msi_version,
+          display_version: msi_display_version,
+        }
+      )
+    end
+
+    #
+    # Write the source file into the staging directory.
+    #
+    # @return [void]
+    #
+    def write_source_file
+      paths = []
+
+      # Remove C:/
+      install_dir = project.install_dir.split('/')[1..-1].join('/')
```

Comment:
I don't think this is relevant any more either.  No directory hierarchy to pre-create.